cli/command/registry/login.go

```diff
@@ -47,6 +50,28 @@ func runLogin(dockerCli command.Cli, opts loginOptions) error {
 	ctx := context.Background()
 	clnt := dockerCli.Client()
 
+	if opts.password != "" {
+		fmt.Fprintf(dockerCli.Err(), "Using --password via the CLI is insecure. Use --password-stdin\n")
+		if opts.passwordStdin {
+			return errors.Errorf("--password and --password-stdin are mutually exclusive")
+		}
+	}
+
+	if opts.passwordStdin {
+		if opts.user == "" {
+			return errors.Errorf("Must provide --username with --password-stdin")
+		}
```

Comment:
If you don't supply a username via --username, it asks via stdin, which then may eat parts of your password, depending on what characters it has in it (e.g. `\n`). Seems better to just disallow it to prevent confusion to me.